src/org/ods/orchestration/usecase/LeVADocumentUseCase.groovy

```diff
@@ -626,6 +625,23 @@ class LeVADocumentUseCase extends DocGenUseCase {
         return uri
     }
 
+    /**
+     * If the risk is associated to technical specification task which is itself associated with a User story,
+     * return the technical specification task
+     * otherwise, return the system requirements
+     * @param risk
+     * @return
+     */
+    private List<Project.JiraDataItem> getRequirements(Project.JiraDataItem risk) {
+        def requirements
+        if (risk.getResolvedTechnicalSpecifications().size() > 0) {
```

Comment:
getResolvedTechnicalSpecifications() is a slow method. You don't want to call it twice.
I suggest you initialize the variable requirements with getResolvedTechnicalSpecification and then use the variable.